test/integration/virtualservice/virtualservice.go

```diff
@@ -0,0 +1,133 @@
+	if err != nil {
+		return err
+	}
+	m.VirtualServices[vs.Name] = vs
+	return nil
+}
+
+func (m *VirtualServiceTest) Update(ctx context.Context, f *framework.Framework, newVS *appmesh.VirtualService, vs *appmesh.VirtualService) (*appmesh.VirtualService, error) {
+	err := f.K8sClient.Patch(ctx, newVS, client.MergeFrom(vs))
+	if err != nil {
+		return nil, err
+	}
+	updatedVS, err := f.VSManager.WaitUntilVirtualServiceActive(ctx, newVS)
+	if err != nil {
+		return nil, err
+	}
+
+	m.VirtualServices[vs.Name] = updatedVS
+	return updatedVS, nil
+}
+
+func (m *VirtualServiceTest) Cleanup(ctx context.Context, f *framework.Framework) {
+	var deletionErrors []error
+
+	for _, vs := range m.VirtualServices {
+		By(fmt.Sprintf("Delete virtual service %s", vs.Name), func() {
+			if err := f.K8sClient.Delete(ctx, vs,
+				client.PropagationPolicy(metav1.DeletePropagationForeground), client.GracePeriodSeconds(0)); err != nil {
+
+				if apierrs.IsNotFound(err) {
+					f.Logger.Info("Virtual service already deleted",
+						zap.String("virtual sercice", vs.Name))
+					return
+				}
+
+				f.Logger.Error("Failed to delete virtual service",
+					zap.String("virtual service", vs.Name),
+					zap.Error(err))
+				deletionErrors = append(deletionErrors, err)
+				return
+			}
+
+			By(fmt.Sprintf("Wait for virtual service to be deleted: %s", vs.Name), func() {
+				if err := f.VSManager.WaitUntilVirtualServiceDeleted(ctx, vs); err != nil {
+					f.Logger.Error("failed to wait virtual service deletion",
+						zap.String("virtual service", vs.Name),
+						zap.Error(err))
+					deletionErrors = append(deletionErrors, err)
+				}
+			})
+			delete(m.VirtualServices, vs.Name)
+		})
+	}
+
+	if m.Namespace != nil {
+		By(fmt.Sprintf("Delete namespace: %s", m.Namespace.Name), func() {
+			if err := f.K8sClient.Delete(ctx, m.Namespace,
+				client.PropagationPolicy(metav1.DeletePropagationForeground), client.GracePeriodSeconds(0)); err != nil {
+
+				if !apierrs.IsNotFound(err) {
+					f.Logger.Error("failed to delete namespace",
+						zap.String("namespace", m.Namespace.Name),
+						zap.Error(err))
+					deletionErrors = append(deletionErrors, err)
+					return
+				}
+			}
+
+			By(fmt.Sprintf("Wait for the namespace to be deleted: %s", m.Namespace.Namespace), func() {
+				if err := f.NSManager.WaitUntilNamespaceDeleted(ctx, m.Namespace); err != nil {
+					f.Logger.Error("failed to wait namespace deletion",
+						zap.String("namespace", m.Namespace.Name),
+						zap.Error(err))
+					deletionErrors = append(deletionErrors, err)
+				}
+			})
+		})
+	}
+
+	for _, err := range deletionErrors {
+		f.Logger.Error("VirtualService clean up failed", zap.Error(err))
+	}
+	Expect(len(deletionErrors)).To(BeZero())
+}
+
+func (m *VirtualServiceTest) CheckInAWS(ctx context.Context, f *framework.Framework, ms *appmesh.Mesh, vs *appmesh.VirtualService) error {
+	var err error
+	// Reconcile may take a while so add re-tries
```

Comment:
Thinking about it, based on our experience from the scale tests the current value of 1 sec might not hold if we scale no.of Virtualservices. How about we adjust "retryWait" based on the scale? If not, we should introduce something in the lines of "initialDelay" that is configurable based on the scale in the test suite before we land here.

Reply:
Initial delay isn't really deterministic so not a fan of that approach :( 

You're right on the scale. Since these aren't scale tests right now, we can adjust the retry wait when we adapt these for scale